### PR TITLE
fix preview output when no output param and type doesn't exist

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

- When a file extension was not provided, the preview endpoint without `output` param would fail as output type would be empty (default was taken from the file extension). This PR fixes by making default output to `jpg` if both output and file type are missing.
- Adds `_APP_STORAGE_PREVIEW_LIMIT` environment variable to configure the image preview file size limit

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

#2770 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

File: app/controllers/api/storage.php [hunk 1]
```diff
@@ -950,9 +950,8 @@
         $cache = new Cache(new Filesystem(APP_STORAGE_CACHE . DIRECTORY_SEPARATOR . 'app-' . $project->getId() . DIRECTORY_SEPARATOR . $bucketId . DIRECTORY_SEPARATOR . $fileId)); // Limit file number or size
         $data = $cache->load($key, 60 * 60 * 24 * 30 * 3/* 3 months */);
 
+        $output = (empty($output)) ? ( empty($type) ? 'jpg' : $type ) : $output;
```
Comment (eldadfux) [inline app/controllers/api/storage.php:953]: Why are we assuming jpeg? What happens if the file is not an image? Maybe using the file mime-type to determine the default would be better.

Comment (lohanidamodar) [reply app/controllers/api/storage.php:953]: If the file is not an image, that part is already calculated above to get a default image preview.

Comment (lohanidamodar) [reply app/controllers/api/storage.php:953]: Using mime/type here to determine will again cause issue if the mime-type is not of an image.

Comment (eldadfux) [reply app/controllers/api/storage.php:953]: Agree, but if it is an image, we can tell if it's jpg, gif, or png.

Comment (lohanidamodar) [reply app/controllers/api/storage.php:953]: If it's an image, `$type` already has the type and it takes `$type` instead of default `jpg` ?

**Review by eldadfux — APPROVED** [review]

File: app/controllers/api/storage.php [hunk 2]
```diff
@@ -944,9 +944,11 @@
         $cache = new Cache(new Filesystem(APP_STORAGE_CACHE . DIRECTORY_SEPARATOR . 'app-' . $project->getId() . DIRECTORY_SEPARATOR . $bucketId . DIRECTORY_SEPARATOR . $fileId)); // Limit file number or size
         $data = $cache->load($key, 60 * 60 * 24 * 30 * 3/* 3 months */);
 
-        if ($data) {
-            $output = (empty($output)) ? $type : $output;
+        if(empty($output)) {
+            $output = empty($type) ? (array_search($mime, $outputs) ?? 'jpg') : $type;
```
Comment (wess) [inline app/controllers/api/storage.php:948]: If reading correctly, we are falling back to jpg for file type?

**Review by Meldiron — APPROVED** [review]

**Comment by TorstenDittmann** [comment]
> This PR is for `App::get('/v1/storage/buckets/:bucketId/files/:fileId/preview')` and we are assuming images only for this, perhaps it should be `:bucketId/images/...` otherwise we can't assume everything is an image, yes?

That would imply that an image is not a file? 😉

**Comment by wess** [comment]
> This PR is for App::get('/v1/storage/buckets/:bucketId/files/:fileId/preview') and we are > assuming images only for this, perhaps it should be :bucketId/images/... otherwise we can't assume everything is an image, yes?


> That would imply that an image is not a file? 

No but `jpg` as the fallback implies it's an image, do we offer preview for, say, a text file? :)

**Comment by lohanidamodar** [comment]
> > This PR is for App::get('/v1/storage/buckets/:bucketId/files/:fileId/preview') and we are > assuming images only for this, perhaps it should be :bucketId/images/... otherwise we can't assume everything is an image, yes?
> 
> > That would imply that an image is not a file?
> 
> No but `jpg` as the fallback implies it's an image, do we offer preview for, say, a text file? :)

Yes for non image files we have a fallback icons used for preview. So everything coming to this point is a file with mime type for some kind of image.

File: app/controllers/api/storage.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (lohanidamodar) [reply app/controllers/api/storage.php:948]: Sure we can,

Comment (lohanidamodar) [reply app/controllers/api/storage.php:948]: yes at this point, what we receive will be image. Non image files will use default logo images for preview.

[the remark this answers is not in the document]
Comment (lohanidamodar) [reply app/controllers/api/storage.php:950]: added comment

**Review by wess — APPROVED** [review]

**Comment by lohanidamodar** [comment]
@TorstenDittmann if you don't have any other concerns let's merge this.

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**